From 415eefb8ff4a1a9a749fe256320e1d1e276fa20a Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 5 Sep 2015 23:49:16 +0200
Subject: [PATCH] Split up --with-db

---
 configure.ac | 41 +++++++++--------------------------------
 1 file changed, 9 insertions(+), 32 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3a4fbe6..d3fdcb3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -251,7 +251,9 @@ AC_ARG_ENABLE(pie, [  --enable-pie            build position-independent executa
 
 AC_ARG_WITH(pam, [  --with-pam              use PAM if present (see PAM.txt before you try this)])
 AC_ARG_WITH(kthread, [  --with-kthread          use kernel threads (on FreeBSD) (not recommended yet)])
-AC_ARG_WITH(db, [  --with-db@<:@=DIR@:>@         use Berkeley DB 3.x @<:@DIR=/usr/local/BerkeleyDB.3.@<:@123@:>@@:>@])
+AC_ARG_WITH(db-include, [  --with-db-include=DIR   set Berkeley DB 4.x include path, e.g. /usr/include/db4.5/])
+AC_ARG_WITH(db-lib, [  --with-db-lib=DIR       set Berkeley DB 4.x library path, /usr/lib64/])
+AC_ARG_WITH(db-name, [  --with-db-name=NAME     set Berkeley DB 4.x library name, e.g. "db-4.5"])
 AC_ARG_WITH(ssl,
 	[  --with-ssl=PATH         Specify path to OpenSSL installation ],
 	[
@@ -264,21 +266,6 @@ AC_ARG_WITH(with_ldap, [  --with-ldap             use OpenLDAP client library])
 dnl AC_ARG_WITH(with_libdspam, [  --with-libdspam         use libdspam mail spam scanning library])
 AC_ARG_WITH(with_gc, [  --with-gc             use the Boehm-Demers-Weiser garbage collection library])
 
-if test "x$with_db" != xno -a "x$with_db" != xyes -a "$with_db"; then
-	db_dir="$with_db"
-	with_db=yes
-else
-	test -f /usr/local/lib/libdb.a -o -f /usr/local/lib/libdb.so \
-	  -o -f /usr/local/lib/libdb4.a -o -f /usr/local/lib/libdb4.so \
-	  && db_dir=/usr/local
-
-	test -d /usr/local/BerkeleyDB.4.1 && db_dir=/usr/local/BerkeleyDB.4.1
-	test -d /usr/local/BerkeleyDB.4.2 && db_dir=/usr/local/BerkeleyDB.4.2
-	test -d /usr/local/BerkeleyDB.4.3 && db_dir=/usr/local/BerkeleyDB.4.3
-	test -d /usr/local/BerkeleyDB.4.4 && db_dir=/usr/local/BerkeleyDB.4.4
-	test -d /usr/local/BerkeleyDB.4.5 && db_dir=/usr/local/BerkeleyDB.4.5
-fi
-
 AC_CANONICAL_HOST
 PTHREAD_DEFS=-D_REENTRANT
 AC_MSG_CHECKING([how to compile with POSIX threads])
@@ -728,28 +715,18 @@ if test "$with_ssl" != "no"; then
 	fi
 fi
 
-if test "x$with_db" != xno; then
-	test "$db_dir" && LDFLAGS="$LDFLAGS -L$db_dir/lib"
-
-	dblib=""
-
-	if test -d "$db_dir/include/db4"; then
-		CPPFLAGS="$CPPFLAGS -I$db_dir/include/db4"
-		dblib="db4"
-	elif test "$db_dir"; then
-		CPPFLAGS="$CPPFLAGS -I$db_dir/include"
-	elif test -d /usr/include/db4; then
-		CPPFLAGS="$CPPFLAGS -I/usr/include/db4"
-		dblib="db4"
-	fi
 
+if true; then
+	LDFLAGS="$LDFLAGS -L${with_db_lib}"
+	CPPFLAGS="$CPPFLAGS -I${with_db_include}"
+	dblib="lib${with_db_name}"
 
-	AC_CHECK_DB([db db-4.1 db-4 db4],
+	AC_CHECK_DB([${with_db_name}],
 		    [
 		     DATABASE=database.c
 		    ],
 		    AC_MSG_ERROR([[Can not locate a suitable Berkeley DB
-				  library.  Use --with-db=PATH to specify
+				  library.  Use --with-db-(include|lib|name)=... to specify
 				  the path]]))
 
 fi
-- 
2.5.0

